backend/plugin/db/elasticsearch/spliter.go

```diff
@@ -1,138 +1,176 @@
 package elasticsearch
 
 import (
-	"errors"
+	"strings"
+
+	"github.com/cockroachdb/errors"
 )
 
-func SplitElasticsearchStatements(statementsStr string) ([]*Statement, error) {
-	var sm StateMachine
-	var statements []*Statement
+func splitElasticsearchStatements(statementsStr string) ([]*statement, error) {
+	var stats []*statement
+	sm := &stateMachine{}
 
 	for idx, c := range statementsStr {
-		statement := sm.transfer(c)
-		if sm.state == StatusError {
-			return nil, errors.New("failed to parse statements")
+		stat, err := sm.transfer(c)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse statements")
 		}
-		if statement != nil {
-			statements = append(statements, statement)
-		} else if idx == len(statementsStr)-1 && isMethodSupported(sm.statement.method) && sm.statement.route != nil {
-			tmpStatement := &Statement{
-				method: sm.statement.method,
-				route:  sm.statement.route,
-			}
-			if sm.statement.queryString != nil && sm.numLeftBrace == 0 {
-				tmpStatement.queryString = sm.statement.queryString
+		// Generate a statement when the maximum parsed content length is reached.
```

Comment:
The code below needs some refactoring for simplicity.

The stat is expected to be nil if err is not nil typically.